pkg/cloudprovider/drift.go

```diff
@@ -73,3 +83,28 @@ func (c *CloudProvider) isAMIDrifted(ctx context.Context, machine *v1alpha5.Mach
 	}
 	return !lo.Contains(lo.Keys(mappedAMIs), instance.ImageID), nil
 }
+
+func (c *CloudProvider) isSecurityGroupDrifted(ec2Instance *instance.Instance, nodeTemplate *v1alpha1.AWSNodeTemplate) bool {
+	securityGroupIds := lo.Map(nodeTemplate.Status.SecurityGroups, func(sg v1alpha1.SecurityGroup, _ int) string { return sg.ID })
```

Comment:
Do you mean in the event that securityGroup is update upstream?

Reply:
The status will reconcile as soon as the `nodetemplate` is updated. The drift check will not happen if the `nodetemplate` gets updated, because drift check will occurs every 5 min

Reply:
I meant in the case that the AWSNodeTemplate Status reconciliation fails, and there's no value in the `nodeTemplate.Status.SecurityGroups` field.